drp.py

```diff
@@ -30,55 +30,42 @@ def base_activity():
         activity['timestamps'] = {'start': start_time}
     return activity
 
-
-# TODO base these on base scope name
-EXT_ICON_MAP = {
-    '.js': 'javascript',
-    '.py': 'python',
-    '.lua': 'lua',
-    '.rb': 'ruby',
-    '.gemspec': 'ruby',
-    '.cr': 'crystal',
-    '.css': 'css',
-    '.html': 'html',
-    '.htm': 'html',
-    '.shtml': 'html',
-    '.xhtml': 'html',
-    '.md': 'markdown',
-    '.mdown': 'markdown',
-    '.markdown': 'markdown',
-    '.markdn': 'markdown',
-    '.cs': 'cs',
-    '.csproj': 'cs',
-    '.cpp': 'cpp',
-    '.php': 'php',
-    '.php3': 'php',
-    '.go': 'go',
-    '.d': 'd',
-    '.c': 'c',
-    '.json': 'json',
-    '.exs': 'elixir',
-    '.ex': 'elixir',
-    '.java': 'java',
-    '.properties': 'java',
-    '.ts': 'typescript',
-    '.sublime-settings': 'json',
-    '.sublime-snippet': 'json',
-    '.sublime-theme': 'json',
-    '.sublime-menu': 'json',
-    '.sublime-commands': 'json',
-    '.sublime-keymap': 'json',
-    '.sublime-mousemap': 'json',
-    '.sublime-build': 'json',
-    '.sublime-macro': 'json',
-    '.sublime-completions': 'json',
-    '.sublime-project': 'json'
+SCOPE_ICON_MAP = {
+    'source.c': 'c',
+    'source.c++': 'cpp',
+    'source.crystal': 'crystal',
+    'source.cs': 'cs',
+    'source.css': 'css',
+    'source.d': 'd',
+    'source.elixir': 'elixir',
+    'source.go': 'go',
+    'source.java': 'java',
+    'source.java-props': 'java',
+    'source.js': 'javascript',
+    'source.json': 'json',
+    'source.lua': 'lua',
+    'source.php': 'php',
+    'source.python': 'python',
+    'source.ruby': 'ruby',
+    'source.ts': 'typescript',
+    'text.html': 'html',
+    'text.html.markdown': 'markdown',
 }
 
-
-def get_icon(ext):
-    return 'lang-%s' % EXT_ICON_MAP.get(ext, "unknown")
-
+def get_icon(main_scope):
+    icon = 'unknown'
+    for scope in yield_subscopes(main_scope):
```

Comment:
You could also be using `sublime.match_selector` here and pick the match with the highest non-zero result.

Reply:
I prefer it like this because it allows me to find greedily the best scope.
Using `match_selector` would require 1. compute the score for all scopes, 2. return the scope with the best score.

Reply:
With the way it is implemented now (after SCOPE_ICON_MAP was split and the source/text stuff removed, your implementation kind of becomes simpler.